src/rules/noUnusedVariableRule.ts

```diff
@@ -33,18 +39,58 @@ class NoUnusedVariablesWalker extends Lint.RuleWalker {
     private skipParameterDeclaration: boolean;
     private skipVariableDeclaration: boolean;
 
+    // after a this.walk is completed, this.reactImport will be set to a NamespaceImport
+    // iff:
+    //
+    // + a react option has been provided to the rule and
+    // + an import of a module that matches one of OPTION_REACT_MODULES is found, to a
+    //   NamespaceImport named OPTION_REACT_NAMESPACE_IMPORT_NAME
+    //
+    // e.g.
+    //
+    // import * as React from "react/addons";
+    //
+    // If reactImport is defined when a walk is completed, we need to have:
+    //
+    // a) seen another usage of React and/or
+    // b) seen a JSX identifier
+    //
+    // otherwise a usage error will be reported
```

Comment:
change this to "variable usage failure"

"error" is ambiguous in JS, so we should stick to the "failure" terminology.